aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -147,7 +151,8 @@ static void upsample_nearest2d_out_cuda_template(
   // upsample_2d_shape_check makes sure input/output tensor is not empty;
   int block_x = std::min<int>(lastPow2(output_width), max_threads);
   int block_y = std::min<int>(lastPow2(output_height), max_threads / block_x);
-  int block_z = std::min<int>(nc, max_threads / block_x / block_y);
+  int block_z =
+      std::min<int>(64, std::min<int>(nc, max_threads / block_x / block_y));
```

Comment:
Can you please query maxThreadDims?  
https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_192d195493a9d36b2d827aaf3ffd89f1e